conans/test/integration/toolchains/intel/test_using_msbuild.py

```diff
@@ -0,0 +1,63 @@
+import os
+import platform
+import textwrap
+import unittest
+
+from conans.client.tools.win import vs_installation_path
+
+from ._base import BaseIntelTestCase
+from ..test_content import create_solution
+
+
+@unittest.skipUnless(platform.system() == "Windows", "msbuild requires Windows")
+class MSBuildIntelTestCase(BaseIntelTestCase):
+    def test_use_msbuild_toolchain(self):
+        create_solution(self.t)
+
+        conanfile_py = textwrap.dedent("""
+            from conans import ConanFile, MSBuild, MSBuildToolchain
+
+            class Library(ConanFile):
+                name = 'foobar'
+                settings = 'os', 'arch', 'compiler', 'build_type'
+                exports_sources = "foobar.vcxproj", "foobar.vcxproj.filters", "foobar.sln", "foobar.hpp", "foobar.cpp"
+
+                def toolchain(self):
+                    tc = MSBuildToolchain(self)
+                    tc.write_toolchain_files()
+
+                def build(self):
+                    msbuild = MSBuild(self)
+                    msbuild.build("foobar.sln")
+        """)
+
+        self.t.save({'conanfile.py': conanfile_py,
+                    'profile': self.profile})
+
+        lib = os.path.join(self.t.current_folder, 'x64', 'Release', 'foobar.lib')
+
+        vs_path = vs_installation_path("15")
+        vcvars_path = os.path.join(vs_path, "VC/Auxiliary/Build/vcvarsall.bat")
+
+        vcvars_cmd = 'set "VSCMD_START_DIR=%%CD%%" && "%s" x64 && ' % vcvars_path
+
+        # Build in the cache
+        self.t.run('create . foobar/version@ --profile:host=profile')
+
+        # Build locally
+
+        build_type = 'Release'
+        platform_arch = 'x64'
+        toolset = 'Intel C++ Compiler 19.1'
+
+        cmd = ('%s msbuild "foobar.sln" /p:Configuration=%s '
+               '/p:Platform=%s /p:PlatformToolset="%s"' %
+               (vcvars_cmd, build_type, platform_arch, toolset))
+
+        self.t.run('install . foobar/version@ --profile:host=profile')
+        self.t.run_command(cmd)
+
+        self.assertTrue(os.path.isfile(lib))
```

Comment:
Lets try to build an executable instead, that can be ran and that outputs the compiler it was used to be built (via preprocessor definitions specific of the compiler). Please check other updated toolchain tests like ``test_msbuild`` and its checks:

```python
self.assertIn("main _MSC_VER1900", client.out)
self.assertIn("main _MSVC_LANG2014", client.out)
```

Reply:
I now build executable and check:
```
self.assertIn("main __INTEL_COMPILER1910", self.t.out)
```